contrib/pxf_fdw/pxf_fragment.c

```diff
@@ -394,22 +374,79 @@ FilterFragmentsForSegment(List *list)
 			ListCell   *to_delete = current;
 
 			if (to_delete->data.ptr_value)
-				free_fragment((FragmentData *) to_delete->data.ptr_value);
+				FreeFragment((FragmentData *) to_delete->data.ptr_value);
 			current = lnext(to_delete);
 			result = list_delete_cell(list, to_delete, previous);
 		}
 	}
 	return result;
 }
 
+/*
+ * Serializes a List of FragmentData such that it can be transmitted
+ * via the interconnect, and the list can be passed from master to
+ * segments
+ */
+List *
+SerializeFragmentList(List *fragments)
+{
+	ListCell   *frag_c = NULL;
+	List	   *serializedFragmentList = NIL;
+	List	   *serializedFragment;
+
+	foreach(frag_c, fragments)
+	{
+		FragmentData *fragment = (FragmentData *) lfirst(frag_c);
+
+		serializedFragment = NIL;
+		serializedFragment = lappend(serializedFragment, makeString(fragment->index));
+		serializedFragment = lappend(serializedFragment, makeString(fragment->source_name));
+		serializedFragment = lappend(serializedFragment, makeString(fragment->fragment_md));
+		serializedFragment = lappend(serializedFragment, makeString(fragment->user_data));
+		serializedFragment = lappend(serializedFragment, makeString(fragment->profile));
+		serializedFragment = lappend(serializedFragment, makeString(psprintf("%d", fragment->fragment_idx)));
+
+		serializedFragmentList = lappend(serializedFragmentList, serializedFragment);
+	}
+
+	return serializedFragmentList;
+}
+
+/*
+ * Deserializes a List of FragmentData received from the interconnect
+ */
+List *
+DeserializeFragmentList(List *serializedFragmentList)
+{
+	ListCell	*frag_c = NULL;
+	List		*fragments = NIL;
+
+	foreach(frag_c, serializedFragmentList)
+	{
+		List *serializedFragment = (List *) lfirst(frag_c);
+
+		FragmentData *fragment = (FragmentData *) palloc0(sizeof(FragmentData));
+		fragment->index = strVal(list_nth(serializedFragment, 0));
+		fragment->source_name = strVal(list_nth(serializedFragment, 1));
+		fragment->fragment_md = strVal(list_nth(serializedFragment, 2));
+		fragment->user_data = strVal(list_nth(serializedFragment, 3));
+		fragment->profile = strVal(list_nth(serializedFragment, 4));
+		fragment->fragment_idx = atoi(strVal(list_nth(serializedFragment, 5)));
```

Comment:
strtol(strVal(list_nth(serializedFragment, 5)), NULL, 10);
Suggest using strtol as fdw upstream also use the function to turn string to int